mito-ai/mito_ai/utils/mito_server_utils.py

```diff
@@ -0,0 +1,85 @@
+from mito_ai.completions.models import MessageType
+from mito_ai.utils.server_limits import update_mito_server_quota
+from tornado.httpclient import HTTPResponse
+import time
+import json
+from typing import Any, Dict, Optional
+from mito_ai.constants import MITO_GEMINI_URL
+from mito_ai.utils.utils import _create_http_client
+
+
+class ProviderCompletionException(Exception):
+    """Custom exception for Mito server errors that converts well to CompletionError."""
+    
+    def __init__(self, error_message: str, provider_name: str = "LLM Provider", error_type: str = "LLMProviderError"):
+        self.error_message = error_message
+        self.provider_name = provider_name
+        self.error_type = error_type
+        
+        # Create user-friendly title and hint
+        self.user_friendly_title = f"{provider_name} Error: {error_message}"
+        self.user_friendly_hint = f"There was a problem with {provider_name}. Try switching to a different model and trying again."
+
+        # Set args[0] for fallback compatibility
+        super().__init__(self.user_friendly_title)
+
+    def __str__(self):
+        return f"{self.provider_name} Error: {self.error_message}"
+
+
+async def get_response_from_mito_server(
```

Comment:
Creating this function was the main refactor done in this PR. We now extract the completions + handle any errors here. Using this function for all of our clients makes sure that we have the same return type for all clients, making it much easier to understand that data types we're working with!